File: SDKs/amazonaws.com/email/2010-12-01/go/pkg/models/operations/postsendtemplatedemail.go
package operations

type PostSendTemplatedEmailActionEnum string

const (
	PostSendTemplatedEmailActionEnumSendTemplatedEmail PostSendTemplatedEmailActionEnum = "SendTemplatedEmail"
)

type PostSendTemplatedEmailVersionEnum string

const (
	PostSendTemplatedEmailVersionEnumTwoThousandAndTen1201 PostSendTemplatedEmailVersionEnum = "2010-12-01"
)

type PostSendTemplatedEmailQueryParams struct {
	Action  PostSendTemplatedEmailActionEnum  `queryParam:"style=form,explode=true,name=Action"`
	Version PostSendTemplatedEmailVersionEnum `queryParam:"style=form,explode=true,name=Version"`
}

type PostSendTemplatedEmailHeaders struct {
	XAmzAlgorithm     *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
}

type PostSendTemplatedEmailRequest struct {
	QueryParams PostSendTemplatedEmailQueryParams
	Headers     PostSendTemplatedEmailHeaders
	Request     []byte `request:"mediaType=text/xml"`
}

type PostSendTemplatedEmailResponse struct {
	Body        []byte
	ContentType string
	StatusCode  int64
}

File: SDKs/amazonaws.com/email/2010-12-01/typescript/src/sdk/models/operations/postsendtemplatedemail.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostSendTemplatedEmailActionEnum {
    SendTemplatedEmail = "SendTemplatedEmail"
}

export enum PostSendTemplatedEmailVersionEnum {
    TwoThousandAndTen1201 = "2010-12-01"
}


export class PostSendTemplatedEmailQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Action" })
  action: PostSendTemplatedEmailActionEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=Version" })
  version: PostSendTemplatedEmailVersionEnum;
}


export class PostSendTemplatedEmailHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class PostSendTemplatedEmailRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: PostSendTemplatedEmailQueryParams;

  @SpeakeasyMetadata()
  headers: PostSendTemplatedEmailHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=text/xml" })
  request?: Uint8Array;
}


export class PostSendTemplatedEmailResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  body?: Uint8Array;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}
